AK/Optional.h

```diff
@@ -32,10 +36,17 @@ struct ConditionallyResultType<false, T> {
     using Type = T;
 };
 
+template<typename Self, typename T>
+struct AddConstIfNeeded {
+    using Type = Conditional<IsConst<RemoveReference<Self>> && !IsConst<T>, AddConst<T>, T>;
```

Comment:
~~🤔 what's the point of this? `AddConst<T const>` is just `T const` anyway~~
This name is really confusing, I think the intent is `CopyConst<RemoveReference<Self>, T>` but to keep `T = U const` as const if `Self` isn't const?

regardless the `&& !IsConst<T>` is unnecessary, AddConst is idempotent for T const.

Reply:
Ah, good looked at `AddConst` and did not see any special handling for `T const`, good to know that it is idempotent anyway
And yes this is supposed to be `CopyConstButDontRemoveItPls`

Reply:
maybe a different name then, `AddConstFrom<F, T>` maybe?
```cpp
template<typename From, typename To>
struct AddConstFrom {
    using Type = Conditional<IsConst<RemoveReference<From>>, AddConst<T>, T>;
};
```

the RemoveReference bit seems a bit odd to me, but probably okay.